Source/cmake/FindGStreamer.cmake

```diff
@@ -77,8 +77,9 @@ macro(FIND_GSTREAMER_COMPONENT _component_prefix _pkgconfig_name _library)
     # `.._INCLUDE_DIRS` variable in this case. So we need to do it here...
     set(_include_dir "${PC_${_component_prefix}_INCLUDEDIR}")
     string(REGEX MATCH "(.*)/gstreamer-1.0" _dummy "${_include_dir}")
-    if ("${CMAKE_MATCH_1}" STREQUAL "")
-        set(${_component_prefix}_INCLUDE_DIRS "${_include_dir}/gstreamer-1.0;${PC_${_component_prefix}_INCLUDE_DIRS}")
+    find_path(_full_path_include_dir NAMES "${_include_dir}/gstreamer-1.0")
```

Comment:
The call to `find_path()` should be inside the `if` block, as it is unneeded otherwise. Also `find_path()` will store the result variable in the cache, so second and subsequent calls to the `FIND_GSTREAMER_COMPONENT()` macro will *not* actually try to find anything because the variable will be already present in the cache—one needs an unique result variable name here (or using CMake 3.21+ which added a `NO_CACHE` flag; but this is not an option right now). Lastly, you don't need to manually check for the `-NOTFOUND` suffix, CMake already does that automatically inside `if` expressions.

I think what you want is something like this:

```cmake
# Populate the list initially from the _INCLUDE_DIRS result variable.
set(${_component_prefix}_INCLUDE_DIRS ${PC_${_component_prefix}_INCLUDE_DIRS})
set(_include_dir "${PC_${_component_prefix}_INCLUDEDIR}")
string(REGEX MATCH "(.*)/gstreamer-1.0" _dummy "${_include_dir}")
if ("${CMAKE_MATCH_1}" STREQUAL "")
    find_path(${_component_prefix}_RESOLVED_INCLUDEDIR NAMES "${_include_dir}/gstreamer-1.0")
    # Only add the resolved path from `_INCLUDEDIR` if found.
    if (${_component_prefix}_RESOLVED_INCLUDEDIR)
        list(APPEND ${_component_prefix}_INCLUDE_DIRS
             "${_component_prefix}_RESOLVED_INCLUDEDIR")
    endif ()
endif ()